src/main/java/org/jclouds/vcloud/director/v1_5/domain/dmtf/Envelope.java

```diff
@@ -81,6 +85,6 @@ public ReferencesType getReferences() {
    }
 
    public VirtualSystem getVirtualSystem() {
-      return virtualSystem;
+      return virtualSystem != null ? virtualSystem : virtualSystemCollection.getVirtualSystems().iterator().next();
```

Comment:
Should we have another method that returns all the `VirtualSystem`s?

Maybe we should be more defensive - for example if `virtualSystemCollection` is also null or we got back an empty VirtualSystemCollection then this would give a `NoSuchElementException`. vCD probably won't do that, but we other areas have seen different things returned in different environments that is impossible to guess in advanced by just looking at the docs.

What do we want to return if there are none? We could do:

    if (virtualStream != null) {
        return virtualSystem;
    } else if (virtualSystemCollection == null || virtualSystemCollection.getVirtualSystems() == null) {
        return null;
    } else {
        return Iterables.getFirst(virtualSystemCollection.getVirtualSystems(), null);
    }